frontend/app.js

```diff
@@ -0,0 +1,81 @@
+const ErrorComponent = {
+  render: () => {
+    return `
+      <section>
+        <h1>Error</h1>
+      </section>
+    `;
+  }
+}
+
+// Pre defined routes 
+const routes = [
+  { path: '/', component: HomeComponent, },
+  { path: '/schema-report-connect-to-db', component: SchemaComponent, },
+  { path: '/schema-report-load-db-dump', component: SchemaComponent, },
+  { path: '/schema-report-import-db', component: SchemaComponent, },
+  { path: '/schema-report-resume-session', component: SchemaComponent, },
+  { path: '/instructions', component: InstructionsComponent, }
+];
+
+// function to fetch browser url
+const parseLocation = () => location.hash.slice(1).toLowerCase() || '/';
+
+// function to find component based on browser url
+const findComponentByPath = (path, routes) => routes.find(r => r.path.match(new RegExp(`^\\${path}$`, 'gm'))) || undefined;
+
+// function to render the html based on path
+const router = () => {
+  const path = parseLocation();
+  const { component = ErrorComponent } = findComponentByPath(path, routes) || {};
+
+  if (path == '/schema-report-connect-to-db' && event.type == 'hashchange') {
+    showSchemaAssessment(event.type);
+  }
+  else if (path == '/schema-report-connect-to-db' && event.type == 'load') {
+    const { component = ErrorComponent } = findComponentByPath(location.hash.slice(1).toLowerCase() || '/', routes) || {};
+    document.getElementById('app').innerHTML = component.render();
+    conversionRateResp = JSON.parse(localStorage.getItem('tableBorderColor'));
+    createSourceAndSpannerTables(JSON.parse(localStorage.getItem('conversionReportContent')));
+    createDdlFromJson(JSON.parse(localStorage.getItem('ddlStatementsContent')));
+    createSummaryFromJson(JSON.parse(localStorage.getItem('summaryReportContent')));
+  }
+  else if (path == '/schema-report-load-db-dump' && event.type == 'load') {
+    // onLoadDatabase(localStorage.getItem('globalDbType'), localStorage.getItem('globalDumpFilePath'), event.type);
+    // getSchemaConversionReportContent(event.type);
+    const { component = ErrorComponent } = findComponentByPath(location.hash.slice(1).toLowerCase() || '/', routes) || {};
+    document.getElementById('app').innerHTML = component.render();
+    conversionRateResp = JSON.parse(localStorage.getItem('tableBorderColor'));
+    createSourceAndSpannerTables(JSON.parse(localStorage.getItem('conversionReportContent')));
+    createDdlFromJson(JSON.parse(localStorage.getItem('ddlStatementsContent')));
+    createSummaryFromJson(JSON.parse(localStorage.getItem('summaryReportContent')));
+  }
+  else if (path == '/schema-report-import-db') {
+    onImport();
+  }
+  else if (path == '/schema-report-resume-session') {
+    resumeSession(localStorage.getItem('driver'), localStorage.getItem('path'), localStorage.getItem('fileName'), localStorage.getItem('sourceDb'), event.type);
+  }
+  else {
+    document.getElementById('app').innerHTML = component.render();
+  }
+};
```

Comment:
Made the required change. But I was unable to pass the params inside component.render() because the params are internally using DOM elements which are still not in the DOM until component.render() will return the html content. So the sequence of flow is like
1. component.render() will return html content in DOM and then we can use internal functions of params which are using these DOM elements.
2. so to avoid this error, I have created new function in which I will pass params like ({{'path':path, 'event':event}) and before calling new function component.render() is called

Reply:
Okay, current implementation is also ok but we can improve bit more later in future with mechanism like create afterRender function for each component which can be called from app.js once innerHtml is assigned by render function, and each component will have its own afterRender function just like render function, so we can move the component related logic inside that afterRender function

Reply:
I have kept the current implementation as of now.. with a little modification in current code.. forgot to add it earlier